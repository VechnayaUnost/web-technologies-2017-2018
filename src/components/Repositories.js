import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { fetchReposRequest } from '../actions';
import ReposList from './ReposList';

class Repositories extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            text: ''
        };
    }

    pressEnter(e) {
        if(e.keyCode === 13 && e.shiftKey === false) 
            this.searchRepos();
    };

    textChange(e) {
        console.log(e.target.value);
        this.setState({
            text: e.target.value
        })
    }

    searchRepos() {
        this.props.fetchReposRequest(this.state.text);
    }

    render() {
        return (
            <div>
                <h1>Repositories</h1>
                <div className='inputs'>
                    <input type='text' onChange={this.textChange.bind(this)} onKeyDown={this.pressEnter.bind(this)} placeholder="Repositories" />
                    <button onClick={this.searchRepos.bind(this)}>Search</button>
                </div>
                <ReposList repositories={this.props.repositories}/>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    console.log(state);
    return {
        repositories: state.repos,
    };
};

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchReposRequest,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(Repositories);