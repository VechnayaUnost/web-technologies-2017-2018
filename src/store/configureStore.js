import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};