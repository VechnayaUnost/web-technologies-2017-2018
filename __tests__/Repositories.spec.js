import React from 'react';
import Repositories from '../src/components/Repositories';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('Repositories component test', () => {
    const mockStore = configureStore([thunk]);
    const initialState = {
        repos: [
            {
                name: 'AvatarSkeleton',
                description: 'Creates Avatar skeleton with some preinstalled options',
                stargazers_count: 321,
                watchers_count: 123,
                language: 'C#',
                fork: true,
                archived: false
            },
            {
                name: 'AvatarMountConnection',
                description: 'Avatar mount connection is easier with this tool',
                stargazers_count: 543,
                watchers_count: 345,
                language: 'Java',
                fork: false,
                archived: true
            },
            {
                name: 'SoldiersBarracks',
                description: 'Barrack project for human soldiers',
                stargazers_count: 2,
                watchers_count: 1,
                language: 'Paper&Pen',
                fork: false,
                archived: false
            }
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('Repositories component rendered succesfully', () => {
        const repositories = mount(
            <Repositories store={store} />
        );
        expect(mountToJson(repositories)).toMatchSnapshot();
    });

    it('Repositories component event triggers', () => {
        const props = {
            fetchReposRequest: jest.fn()
        }
        const repositories = mount(
            <Repositories store={store} {...props}/>
        );
        repositories.find('input').simulate('change', { target: { value: 'react' } });
        repositories.find('input').simulate('keyDown', { key: 'Enter', keyCode: 13, which: 13, shiftKey: false });
        repositories.setState({text: 'meow'});
        repositories.find('button').simulate('click');
        repositories.instance().props.fetchReposRequest(repositories.find('button').simulate('click'));
        expect(props.fetchReposRequest).toHaveBeenCalled();
    });
});