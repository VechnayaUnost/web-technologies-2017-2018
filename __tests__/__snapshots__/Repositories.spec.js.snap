// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repositories component test Repositories component rendered succesfully 1`] = `
<Connect(Repositories)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Repositories
    fetchReposRequest={[Function]}
    repositories={
      Array [
        Object {
          "archived": false,
          "description": "Creates Avatar skeleton with some preinstalled options",
          "fork": true,
          "language": "C#",
          "name": "AvatarSkeleton",
          "stargazers_count": 321,
          "watchers_count": 123,
        },
        Object {
          "archived": true,
          "description": "Avatar mount connection is easier with this tool",
          "fork": false,
          "language": "Java",
          "name": "AvatarMountConnection",
          "stargazers_count": 543,
          "watchers_count": 345,
        },
        Object {
          "archived": false,
          "description": "Barrack project for human soldiers",
          "fork": false,
          "language": "Paper&Pen",
          "name": "SoldiersBarracks",
          "stargazers_count": 2,
          "watchers_count": 1,
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div>
      <h1>
        Repositories
      </h1>
      <div
        className="inputs"
      >
        <input
          onChange={[Function]}
          onKeyDown={[Function]}
          placeholder="Repositories"
          type="text"
        />
        <button
          onClick={[Function]}
        >
          Search
        </button>
      </div>
      <ReposList
        repositories={
          Array [
            Object {
              "archived": false,
              "description": "Creates Avatar skeleton with some preinstalled options",
              "fork": true,
              "language": "C#",
              "name": "AvatarSkeleton",
              "stargazers_count": 321,
              "watchers_count": 123,
            },
            Object {
              "archived": true,
              "description": "Avatar mount connection is easier with this tool",
              "fork": false,
              "language": "Java",
              "name": "AvatarMountConnection",
              "stargazers_count": 543,
              "watchers_count": 345,
            },
            Object {
              "archived": false,
              "description": "Barrack project for human soldiers",
              "fork": false,
              "language": "Paper&Pen",
              "name": "SoldiersBarracks",
              "stargazers_count": 2,
              "watchers_count": 1,
            },
          ]
        }
      >
        <ul>
          <Repo
            repo={
              Object {
                "archived": false,
                "description": "Creates Avatar skeleton with some preinstalled options",
                "fork": true,
                "language": "C#",
                "name": "AvatarSkeleton",
                "stargazers_count": 321,
                "watchers_count": 123,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  AvatarSkeleton
                </h3>
                <p>
                  Creates Avatar skeleton with some preinstalled options
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  C#
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  321
                </span>
              </div>
            </li>
          </Repo>
          <Repo
            repo={
              Object {
                "archived": true,
                "description": "Avatar mount connection is easier with this tool",
                "fork": false,
                "language": "Java",
                "name": "AvatarMountConnection",
                "stargazers_count": 543,
                "watchers_count": 345,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  AvatarMountConnection
                </h3>
                <p>
                  Avatar mount connection is easier with this tool
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  Java
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  543
                </span>
              </div>
            </li>
          </Repo>
          <Repo
            repo={
              Object {
                "archived": false,
                "description": "Barrack project for human soldiers",
                "fork": false,
                "language": "Paper&Pen",
                "name": "SoldiersBarracks",
                "stargazers_count": 2,
                "watchers_count": 1,
              }
            }
          >
            <li
              className="repo"
            >
              <div
                className="repo-description"
              >
                <h3>
                  SoldiersBarracks
                </h3>
                <p>
                  Barrack project for human soldiers
                </p>
              </div>
              <div
                className="repo-stars-lang"
              >
                <span
                  className="repo-lang"
                >
                  Language: 
                  Paper&Pen
                </span>
                <span
                  className="repo-stars"
                >
                  Stars: 
                  2
                </span>
              </div>
            </li>
          </Repo>
        </ul>
      </ReposList>
    </div>
  </Repositories>
</Connect(Repositories)>
`;
